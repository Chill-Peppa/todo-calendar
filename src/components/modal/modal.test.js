import renderer from 'react-test-renderer';
import Modal from './modal';

import { render, screen, fireEvent } from '@testing-library/react';

//мокаем работу createPortal
jest.mock('react-dom', () => ({
  ...jest.requireActual('react-dom'),
  createPortal: (children) => children,
}));

const onClose = jest.fn();
const title = 'test title';
const ChildComponent = () => <div>Child Component</div>;

it('should correctly render Modal', () => {
  const tree = renderer
    .create(
      <Modal onClose={onClose} title={title}>
        <ChildComponent />
      </Modal>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('should close modal', () => {
  render(
    <Modal onClose={onClose} title={title}>
      <ChildComponent />
    </Modal>,
  );

  const closeModalButton = screen.getByTestId('close-modal');
  fireEvent.click(closeModalButton);

  expect(onClose).toHaveBeenCalled();
});
