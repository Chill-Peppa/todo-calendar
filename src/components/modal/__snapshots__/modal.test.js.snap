// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Modal 1`] = `
Array [
  <div
    className="modal"
  >
    <div
      className="modal__header"
    >
      <h2
        className="modal__title"
      >
        test title
      </h2>
      <button
        className="modal__close"
        data-testid="close-modal"
        onClick={[MockFunction]}
        type="button"
      >
        <img
          alt="close icon"
          className="modal__close-icon"
          src="close.svg"
        />
      </button>
    </div>
    <div>
      Child Component
    </div>
  </div>,
  <div
    className="modal__background"
    data-testid="modal-background"
    onClick={[Function]}
  />,
]
`;
